bin/vdm

```diff
@@ -67,8 +71,43 @@ try:
     sys.path.insert(0, os.path.join(base_dir, 'lib/python', 'vdm'))
     from voltcli import runner
     from server import HTTPListener
+
 # Be selective about exceptions to avoid masking load-time library exceptions.
 except (IOError, OSError, ImportError), e:
     sys.stderr.write('Exception (%s): %s\n' % (e.__class__.__name__, str(e)))
     sys.exit(1)
-HTTPListener.main(runner, HTTPListener, 8000)
+
+def main():
+    parser = OptionParser(usage="usage: %prog [options] filepath",
+                          version="%prog 1.0")
+    parser.add_option("-p", "--path",
+                  action="store", type="string", dest="filepath")
+    (options, args) = parser.parse_args()
+
+    return options.filepath
+
+
+if __name__ == '__main__':
+  path = main()
+
+app_root = os.path.dirname(os.path.abspath(__file__))
+os.chdir(os.path.normpath(app_root + os.sep + os.pardir) + '/lib/python/vdm')
+os.getcwd()
+
+if path is None:
+    home = expanduser("~")
+    # path = home + '.vdm' if home.endswith('/') else home + '/' + '.vdm'
+    path = os.path.join(home, '.vdm')
+
+if os.path.isdir(str(path)):
+    if os.access(str(path), os.W_OK):
+        HTTPListener.main(runner, HTTPListener, 8000, path)
+    else:
+        print 'Error: There is no permission to create file in this folder. ' \
+              'Unable to start VDM.'
```

Comment:
If the application is not able to find the specified path or cannot write to it, it will display the error message in command line and the VDM application will not start.